internal/notification/handlers/notification_worker.go

```diff
@@ -419,27 +462,43 @@ func (w *NotificationWorker) rollbackToSavepoint(ctx context.Context, tx *sql.Tx
 		Error("could not rollback to savepoint for notification event")
 }
 
-func (w *NotificationWorker) searchEvents(ctx context.Context, tx *sql.Tx, retry bool) ([]eventstore.Event, error) {
+func (w *NotificationWorker) searchEvents(ctx context.Context, tx *sql.Tx, retry bool) ([]eventstore.Event, bool, error) {
+	var events []eventstore.Event
+	var hasMore = false
+	var err error
+
 	if retry {
-		return w.searchRetryEvents(ctx, tx)
+		events, err = w.searchRetryEvents(ctx, tx)
+	} else {
+		// query events and lock them for update (with skip locked)
+		searchQuery := eventstore.NewSearchQueryBuilder(eventstore.ColumnsEvent).
+			LockRowsDuringTx(tx, eventstore.LockOptionSkipLocked).
+			// Messages older than the MaxTTL, we can be ignored.
+			// The first attempt of a retry might still be older than the TTL and needs to be filtered out later on.
+			CreationDateAfter(w.now().Add(-1*w.config.MaxTtl)).
+			Limit(uint64(w.config.BulkLimit)+1). // NB: +1 to check if there are more events
+			AddQuery().
+			AggregateTypes(notification.AggregateType).
+			EventTypes(notification.RequestedType).
+			Builder().
+			ExcludeAggregateIDs().
+			EventTypes(notification.RetryRequestedType, notification.CanceledType, notification.SentType).
```

Comment:
Additionally, looks like adding the AggregateType filter here actually made performance _worse_ in production settings. The `es_active_indexes` index it was using previously was actually faster than using the `es_projection` index (which lacks created_at)